src/ServiceControl/Recoverability/Retrying/Infrastructure/ReturnToSenderDequeuer.cs

```diff
@@ -349,21 +351,9 @@ public void Stop()
 
             public void StopInternal()
             {
-                if (Interlocked.Exchange(ref disposeSignaled, 1) != 0)
-                {
-                    return;
-                }
-
-                try
-                {
-                    _realDequeuer.Stop();
-                }
-                catch (Exception)
+                lock (startStopLock)
                 {
-                    // Making build go green.
-                    var r = 1 + 1;
-                    Interlocked.Increment(ref r);
-                    // We are shutting down, race condition can result in an exception in the real dequeuer.
+                    realDequeuer.Stop();
```

Comment:
Previous we always did a catch, can the Stop() still throw something? If so, should this be gracefully handled?

Reply:
Well, if `Stop` throws then what is the condition of the underlying transport? Would we be able to `Start` it again? I guess that if `Stop` throws then we can't make any assumptions on the state of the receiver. It may as well be in an invalid state.

Based on that I'd rather kill SC if `Stop` throws by triggering a critical action.